debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/junit/MongoDbDatabaseProvider.java

```diff
@@ -45,15 +53,19 @@ public static MongoDbReplicaSet dockerReplicaSet() {
      * @return MongoDb Replica set
      */
     public static MongoDbReplicaSet dockerAuthReplicaSet() {
-        return dockerReplicaSetBuilder().authEnabled(true).build();
+        var replicaSet = dockerReplicaSetBuilder().authEnabled(true);
+        if (NETWORK != null) {
+            replicaSet.network(NETWORK);
+        }
+        return replicaSet.build();
```

Comment:
There is literary no reason to touch this.

Reply:
There is no way when using this **** static abstraction layer to customize network. It's okay to have this layer, but even jumping out here into the MongoDbDeployment and back into those methods is not good. This is just a way to not throw away that layer and rewrite everything with such a PR. Please file a Jira for cleaning up this abstraction layer.  It is working for now and we are only talking about test code.